components/ActionSheet/tests/unit/MDCActionSheetTest.m

```diff
@@ -89,4 +89,121 @@ - (void)testTitleAndMessageColorWhenMessageSetFirst {
   XCTAssertEqualObjects(messageColor, expectedMessageColor);
 }
 
+- (void)testTitleAndMessageCustomRGBColor {
+  // Given
+  self.actionSheet.title = @"Test title";
+  self.actionSheet.message = @"Test message";
+
+  // When
+  UIColor *titleColor = [UIColor colorWithRed:0.5f green:0.7f blue:0.9f alpha:1.0f];
+  UIColor *messageColor = [UIColor colorWithRed:0.9f green:0.7f blue:0.5f alpha:1.0f];
+
+  self.actionSheet.titleTextColor = titleColor;
+  self.actionSheet.messageTextColor = messageColor;
+
+  // Then
+  XCTAssertEqualObjects(self.actionSheet.header.titleLabel.textColor, titleColor);
+  XCTAssertEqualObjects(self.actionSheet.header.messageLabel.textColor, messageColor);
+}
+
+- (void)testTitleAndMessageCustomHSBColor {
+  // Given
+  self.actionSheet.title = @"Test title";
+  self.actionSheet.message = @"Test message";
+
+  // When
+  UIColor *titleColor = [UIColor colorWithHue:0.5f saturation:0.7f brightness:0.9f alpha:1.0f];
+  UIColor *messageColor = [UIColor colorWithHue:0.9f saturation:0.7f brightness:0.5f alpha:1.0f];
+
+  self.actionSheet.titleTextColor = titleColor;
+  self.actionSheet.messageTextColor = messageColor;
+
+  // Then
+  XCTAssertEqualObjects(self.actionSheet.header.titleLabel.textColor, titleColor);
+  XCTAssertEqualObjects(self.actionSheet.header.messageLabel.textColor, messageColor);
+}
+
+- (void)testTitleAndMessageCustomColor {
+  // Given
+  self.actionSheet.title = @"Test title";
+  self.actionSheet.message = @"Test message";
+
+  // When
+  UIColor *titleColor = [UIColor.blackColor colorWithAlphaComponent:0.3f];
+  UIColor *messageColor = [UIColor.blackColor colorWithAlphaComponent:0.2f];
+
+  self.actionSheet.titleTextColor = titleColor;
+  self.actionSheet.messageTextColor = messageColor;
+
+  // Then
+  XCTAssertEqualObjects(self.actionSheet.header.titleLabel.textColor, titleColor);
+  XCTAssertEqualObjects(self.actionSheet.header.messageLabel.textColor, messageColor);
+}
+
+- (void)testTitleAndMessageColorCorrectAlpha {
+  // Given
+  self.actionSheet.title = @"Test title";
+  self.actionSheet.message = @"Test message";
+  UIColor *titleColor = UIColor.blackColor;
+  UIColor *messageColor = UIColor.blackColor;
+
+  // When
+  UIColor *titleColorChangeAlpha = [titleColor colorWithAlphaComponent:0.6f];
+  UIColor *messageColorChangeAlpha = [messageColor colorWithAlphaComponent:0.5f];
+
+  self.actionSheet.titleTextColor = titleColorChangeAlpha;
+  self.actionSheet.messageTextColor = messageColorChangeAlpha;
+
+  // Then
+  BOOL titleEqual = [titleColor isEqual:self.actionSheet.header.titleLabel.textColor];
+  BOOL messageEqual = [messageColor isEqual:self.actionSheet.header.messageLabel.textColor];
+  XCTAssertFalse(titleEqual);
+  XCTAssertFalse(messageEqual);
```

Comment:
Tip: I believe that if these assertions fail then they'll output the code of the assertion, which can help when reading the logs after a failure. In this regard it's typically better to include the logic in the assertion directly.

Reply:
Done